### ISSUE TITLE
Cascade Delete for Relational Items

### ISSUE THREAD

**Issue by benhaynes** [directus/next#594]
When you delete an item, we should do a check if it is referenced elsewhere relationally, and if so, we should either NULLify or delete the relationship. Ideally all of this is within an API parameter that can be set on a case-by-case basis.

* **Example 1**: You delete a file, and therefore should NULLify all M2O references to that file (Ref: Ref: https://github.com/directus/next/issues/529)
* **Example 2**: You delete an item, and therefore should delete its O2M children translations (Ref: https://github.com/directus/next/issues/593)

**Comment by rijkvanzanten** [comment]
I think:

By default, the delete request will return an error stating that there are related items, and that the delete couldn't go through. Then we can add a query param, that is `?cascade=delete` or `?cascade=nullify`

**Comment by Nitwel** [comment]
What about warning that by deleting this item, references get modified and then you can choose between `set to null`, `delete` and `cancel`. I would also be helpful, to at least see some relations shown, maybe all in another modal?

**Comment by benhaynes** [comment]
Yeah, I like it. The option is nice to have... and it would be even nicer to have a listing of all items in those relationships. Not sure if we want to get so granular as to allow setting delete/nullify per item though.

Might be nice to have a dedicated endpoint for fetching everywhere an item is used relationally. Would be nice to have in the Item Detail Sidebar... _"All the places this item is linked"_
